install.sh

```diff
@@ -81,70 +81,78 @@ generate_taildrop_script() {
   cat << 'EOF' > "${taildrop_script}"
 #!/usr/bin/env bash
 
-#!/usr/bin/env bash
-
-# Main function
+# Simple wrapper around the tailscale CLI to send files to a device.
 main() {
-  # Use associative array to store device status to simplify logic
-  declare -A device_status_map
+  # Get the status of all devices
   local status_output
   status_output=$(tailscale status)
-  local friendly_name
-  local device_list=()
+  declare -A device_status_map
 
-  # Parse and fill associative array with device status
   local line
   while read -r line; do
+    local friendly_name
     friendly_name=$(echo "${line}" | awk '{print $2}')
     if [[ -n "${friendly_name}" ]]; then
-      # Directly check connectivity and assign status
-      if tailscale ping --timeout=0.01s "${friendly_name}" &>/dev/null; then
-        device_status_map["${friendly_name}"]=on
-      else
-        device_status_map["${friendly_name}"]=off
-      fi
+      # Assume device is online. To be checked individually later
+      device_status_map["${friendly_name}"]="unknown"
     fi
   done <<< "${status_output}"
 
-  # Prepare device list for dialog, marking status
+  # Prepare the device list for the dialog
+  local device_list=()
   local name
   for name in "${!device_status_map[@]}"; do
-    device_list+=("${name}" "${name}" "${device_status_map[${name}]}")
+    device_list+=("${name}" "${name}" "off") # Initial state is off. This is updated on the actual check
   done
 
-  # Use kdialog to choose a device
+  # Let the user select a device
   local chosen_device
   chosen_device=$(kdialog --title 'Taildrop' --radiolist "Choose Device" "${device_list[@]}" --geometry 200x50)
-
   if [[ -z "${chosen_device}" ]]; then
     echo "No device selected."
     exit 1
   fi
 
-  # Re-check if device is connected before proceeding
-  if tailscale ping --timeout=0.01s "${chosen_device}" &>/dev/null; then
-    local list_names=""
-    # Process each file provided as argument
-    local file
-    for file in "$@"; do
-      if [[ -f "${file}" ]]; then # Ensure it's a file
-        tailscale file cp "${file}" "${chosen_device}":
-        # Append file name to list for notification
+  # Check if the chosen device is online
+  if ! tailscale ping --timeout=1s "${chosen_device}" &>/dev/null; then
```

Comment:
The only change I would make is to change this to change this command from `if ! tailscale ping --timeout=1s "${chosen_device}" &>/dev/null; then` to `if ! tailscale ping --c 1 "${chosen_device}" &>/dev/null; then`. When I was working on the code originally, I couldn't use the c  flag to send 1 ping, but it seems to be working now. I only used the timeout flag as a workaround to maintain that 1 ping limit.